v2/src/main/scala/com/snowflake/snowpark/Column.scala

```diff
@@ -413,19 +445,29 @@ case class Column(override private[snowpark] val expr: Expr) extends ExprNode wi
    * @group op
    * @since 0.1.0
    */
-  def between(lowerBound: Column, upperBound: Column)(implicit src: SrcPositionInfo): Column = {
-    (this >= lowerBound)(src) && (this <= upperBound)(src)
-  }
+  def between(lowerBound: Column, upperBound: Column)(implicit src: SrcPositionInfo): Column =
+    createColumn(
```

Comment:
@sfc-gh-ryhuang I fixed an issue here. We have a dedicated Representation of `Between`.